### PR TITLE
daemon: Check return value after reading from channel

### LINKED ISSUE #1168: abrtd: check a return value of g_io_channel_read_chars()
We should check a return value of g_io_channel_read_chars() in function handle_signal_cb() in abrtd.c.

### PR DESCRIPTION
**Pull request by mgrabovsky**
Closes #1168

### CODE REVIEW

File: src/daemon/abrtd.c [hunk 1]
```diff
@@ -435,7 +435,24 @@ static gboolean handle_signal_cb(GIOChannel *gio, GIOCondition condition, gpoint
...
+    if (stat == G_IO_STATUS_ERROR)
+    {
+        /* An error occurred on the channel. Report it and quit. */
+        error_msg_and_die(_("Can't read from gio channel: '%s'"),
+                error ? error->message : "");
+    }
+    if (stat == G_IO_STATUS_AGAIN)
+    {
+        /* No new data available. Continue as usual. */
+        return TRUE;
```
Comment (ernestask) [inline src/daemon/abrtd.c]: Could you use `G_SOURCE_{CONTINUE,REMOVE}` instead? Those are a bit easier to understand at a glance.

File: src/daemon/abrtd.c [hunk 2]
```diff
@@ -435,7 +435,24 @@ static gboolean handle_signal_cb(GIOChannel *gio, GIOCondition condition, gpoint
 {
     uint8_t signo;
     gsize len = 0;
-    g_io_channel_read_chars(gio, (void*) &signo, 1, &len, NULL);
+    GError *error = NULL;
+    GIOStatus stat = g_io_channel_read_chars(gio, (void*) &signo, 1, &len, &error);
+    if (stat == G_IO_STATUS_ERROR)
+    {
+        /* An error occurred on the channel. Report it and quit. */
+        error_msg_and_die(_("Can't read from gio channel: '%s'"),
```
Comment (ernestask) [inline src/daemon/abrtd.c]: Oh, right, we don’t want this to be translatable, since it’s not user-facing.

File: src/daemon/abrtd.c [hunk 3]
```diff
@@ -435,7 +435,24 @@ static gboolean handle_signal_cb(GIOChannel *gio, GIOCondition condition, gpoint
...
+                error ? error->message : "");
```
Comment (ernestask) [inline src/daemon/abrtd.c]: Can the error ever be null if the status indicates there was an error?

Comment (mgrabovsky) [reply src/daemon/abrtd.c]: I believe it is possible, at least in theory. The way some checks in `g_io_channel_read_chars()` are [implemented](https://github.com/GNOME/glib/blob/c8692fffe0723aee2bc6710590b5e70ef841b5cc/glib/giochannel.c#L1992) doesn't seem to preclude `error` being `NULL` even after a failure – for instance, when the channel is not readable. (Which _should not_ happen, but we better be safe than segfault-sorry, right?)

File: src/daemon/abrtd.c (hunk 2 above)
Comment (mgrabovsky) [reply src/daemon/abrtd.c]: Will fix.

File: src/daemon/abrtd.c (hunk 3 above)
Comment (ernestask) [reply src/daemon/abrtd.c]: Yeah, that makes sense.

**Comment by ernestask** [comment]
Feel free to merge after pushing.

**Pull request merged**